pkg/guardian/redis_conf_store.go

```diff
@@ -460,14 +685,40 @@ func (rs *RedisConfStore) FetchRouteRateLimits() (map[url.URL]Limit, error) {
 	return res, nil
 }
 
+func (rs *RedisConfStore) ApplyGlobalRateLimitConfig(config GlobalRateLimitConfig) error {
+	configBytes, err := json.Marshal(config)
+	if err != nil {
+		return fmt.Errorf("error marshaling json: %v", err)
+	}
+	pipe := rs.redis.TxPipeline()
+	pipe.Set(redisGlobalRateLimitConfigKey, string(configBytes), 0)
+	pipe.Set(redisUseDeprecatedLimitKey, "false", 0)
+
+	_, err = pipe.Exec()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
 func (rs *RedisConfStore) GetLimit() Limit {
 	rs.conf.RLock()
 	defer rs.conf.RUnlock()
+	if rs.conf.useDeprecatedLimit {
+		return rs.conf.limit
+	}
+	return rs.conf.globalRateLimit.Spec.Limit
```

Comment:
rs.conf.globalRateLimit and .Spec are value fields, and are initialized when the RedisConfStore is created, so these shouldn't be nil. Related to my comment on the GetRouteRateLimit conversation, it might be good to change the GetLimit interface to allow for the non-existence of a global rate limit instead of needing to initialize a default value.